planner/cascades/implementation_rules.go

```diff
@@ -406,13 +406,7 @@ type ImplHashJoinBuildLeft struct {
 
 // Match implements ImplementationRule Match interface.
 func (r *ImplHashJoinBuildLeft) Match(expr *memo.GroupExpr, prop *property.PhysicalProperty) (matched bool) {
-	join := expr.ExprNode.(*plannercore.LogicalJoin)
-	switch join.JoinType {
-	case plannercore.SemiJoin, plannercore.AntiSemiJoin, plannercore.LeftOuterSemiJoin, plannercore.AntiLeftOuterSemiJoin:
-		return false
-	default:
-		return prop.IsEmpty()
-	}
+	return prop.IsEmpty()
```

Comment:
Why remove this? I think we are supposed to match the join type here.

Reply:
Why? For all hash join, we can't guarantee any property.

Reply:
`HashJoinBuildLeft` only works for Inner/LeftOuter/RightOuter joins. So we can just reject other join types without checking the property.